Content.Shared/Posters/SharedPosterSystem.cs

```diff
@@ -0,0 +1,221 @@
+    [Dependency] private readonly FoldableSystem _foldableSystem = default!;
+    [Dependency] private readonly INetManager _net = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<PosterComponent, AfterInteractEvent>(OnInteract);
+        SubscribeLocalEvent<PosterComponent, PosterPlacingDoAfterEvent>(OnPlacingDoAfter);
+        SubscribeLocalEvent<PosterComponent, PosterRemovingDoAfterEvent>(OnRemovingDoAfter);
+        SubscribeLocalEvent<PosterComponent, GetVerbsEvent<UtilityVerb>>(OnPlacingVerb);
+        SubscribeLocalEvent<PosterComponent, GetVerbsEvent<AlternativeVerb>>(OnRemoveVerb);
+    }
+
+    private void OnInteract(Entity<PosterComponent> poster, ref AfterInteractEvent args)
+    {
+        if (!args.CanReach || args.Target == null)
+            return;
+
+        if (CanPlacePoster(poster, args.Target.Value))
+            StartPlacing(poster, args.Target.Value, args.User);
+    }
+
+    private void StartPlacing(Entity<PosterComponent> poster, EntityUid target, EntityUid user)
+    {
+        var comp = poster.Comp;
+
+        var ev = new PosterPlacingDoAfterEvent();
+
+        var doAfterargs = new DoAfterArgs(EntityManager, user, comp.PlacingTime, ev, poster, target, poster)
+        {
+            BreakOnMove = true,
+            NeedHand = true,
+            BreakOnHandChange = true
+        };
+
+        _audioSystem.PlayPredicted(comp.PlacingSound, poster, user);
+
+        if (_doAfter.TryStartDoAfter(doAfterargs))
+            CreatePlaceEffect(poster, target);
+    }
+
+    /// <summary>
+    /// Creating poster place effect on server side
+    /// </summary>
+    private void CreatePlaceEffect(Entity<PosterComponent> poster, EntityUid target)
+    {
+        if (!_net.IsServer)
+            return;
+
+        var comp = poster.Comp;
+        comp.EffectEntity = Spawn(comp.PlacingEffect, Transform(target).Coordinates);
+    }
+
+    private void DeletePlaceEffect(Entity<PosterComponent> poster)
+    {
+        if (!_net.IsServer)
+            return;
+
+        var comp = poster.Comp;
+
+        QueueDel(comp.EffectEntity);
+    }
+
+    private void OnPlacingDoAfter(Entity<PosterComponent> poster, ref PosterPlacingDoAfterEvent args)
+    {
+        // Delete effect in case doafter was cancelled
+        if (args.Cancelled || args.Handled)
+        {
+            DeletePlaceEffect(poster);
+            return;
+        }
+
+        if (args.Target == null)
+            return;
+
+        // Getting foldable comp to unfold poster
+        if (!TryComp<FoldableComponent>(poster, out var foldable))
+            return;
+
+        // Moving entity to the target coords
+        var xform = Transform(poster);
+        var parentXform = Transform(args.Target.Value);
+
+        _transformSystem.SetCoordinates(poster, xform, parentXform.Coordinates, rotation: Angle.Zero);
+        _transformSystem.SetParent(poster, args.Target.Value);
```

Comment:
Wall's deletion or deconstruction will delete the poster because of this line.
I think that in case of wall deletion, poster should fall off the wall back in it's folded state with popup
Besides, I think poster must be anchored on placing and unachored on removing, so things like gravity anomaly or singularity will not rip posters off the wall